File: signage/src/api/weather.ts
export interface WeatherDay {
  date: string; // "YYYY-MM-DD" (JST)
  dayOffset: number; // 0=今日, 1=明日, ...
  weatherCode: number; // 気象庁 天気予報用テロップ番号（1xx 晴 / 2xx 曇 / 3xx 雨 / 4xx 雪）
  description: string;
  tempMin: number | null; // 気象庁が値を持たない場合は null
  tempMax: number | null;
  tempCurrent: number | null; // アメダス現在気温（今日のみ）
  pop: number; // 降水確率 0..1
}

export interface WeatherData {
  days: WeatherDay[];
  fetchedAt: string | null;
}

interface WeatherResponse {
  days: WeatherDay[];
  fetchedAt: string | null;
}

const API_BASE_URL =
  import.meta.env.VITE_API_BASE_URL || "http://127.0.0.1:8080/api";

export async function fetchWeather(): Promise<WeatherData> {
  const res = await fetch(`${API_BASE_URL}/signage/weather`);
  if (!res.ok) {
    throw new Error(`Failed to fetch weather: ${res.status}`);
  }
  const data: WeatherResponse = await res.json();
  return { days: data.days, fetchedAt: data.fetchedAt };
}

File: edge/internal/server/weather.go
package server

import (
	"encoding/json"
	"log"
	"net/http"
)

type weatherDayDTO struct {
	Date        string   `json:"date"`
	DayOffset   int      `json:"dayOffset"`
	WeatherCode int      `json:"weatherCode"`
	Description string   `json:"description"`
	TempMin     *float64 `json:"tempMin"`
	TempMax     *float64 `json:"tempMax"`
	TempCurrent *float64 `json:"tempCurrent"`
	Pop         float64  `json:"pop"`
}

type weatherResponse struct {
	Days      []weatherDayDTO `json:"days"`
	FetchedAt *string         `json:"fetchedAt"`
}

func (s *Server) handleGetWeather(w http.ResponseWriter, r *http.Request) {
	days, fetchedAt, err := s.weather.List(r.Context())
	if err != nil {
		log.Printf("weather list: %v", err)
		http.Error(w, `{"error":"internal error"}`, http.StatusInternalServerError)
		return
	}

	res := weatherResponse{
		Days: make([]weatherDayDTO, 0, len(days)),
	}
	for _, d := range days {
		res.Days = append(res.Days, weatherDayDTO{
			Date:        d.Date,
			DayOffset:   d.DayOffset,
			WeatherCode: d.WeatherCode,
			Description: d.Description,
			TempMin:     d.TempMin,
			TempMax:     d.TempMax,
			TempCurrent: d.TempCurrent,
			Pop:         d.Pop,
		})
	}
	if fetchedAt != nil {
		ts := fetchedAt.UTC().Format("2006-01-02T15:04:05.000Z")
		res.FetchedAt = &ts
	}

	w.Header().Set("Content-Type", "application/json")
	_ = json.NewEncoder(w).Encode(res)
}
